docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/FeatureSelection/SelectFeaturesBasedOnMutualInformation.cs

```diff
@@ -0,0 +1,129 @@
+            var loader = mlContext.Data.CreateTextLoader(
+                columns: new[]
+                    {
+                        new TextLoader.Column("Label", DataKind.Boolean, 0),
+                        new TextLoader.Column("GroupA", DataKind.Single, new [] { new TextLoader.Range(1, 3) }),
+                        new TextLoader.Column("GroupB", DataKind.Single, new [] { new TextLoader.Range(4, 6) }),
+                        new TextLoader.Column("GroupC", DataKind.Single, new [] { new TextLoader.Range(7, 9) }),
+                    },
+                hasHeader: true
+            );
+
+            // Then, we use the loader to load the data as an IDataView.
+            var data = loader.Load(dataFilePath);
+
+            Console.WriteLine("Contents of column 'GroupB'");
+            PrintDataColumn(data, "GroupB");
+            // 5 7 10
+            // 1 2 2
+            // 1 3 4
+            // 3 2 1
+
+            Console.WriteLine("Contents of column 'GroupC'");
+            PrintDataColumn(data, "GroupC");
+            // 3 2 1
+            // 3 1 1
+            // 3 7 1
+            // 3 1 1
+
+            // Second, we define the transformations that we apply on the data. Remember that an Estimator does not transform data
+            // directly, but it needs to be trained on data using .Fit(), and it will output a Transformer, which can transform data.
+
+            // We define a MutualInformationFeatureSelectingEstimator that selects the top k slots in a feature 
+            // vector based on highest mutual information between that slot and a specified label. Notice that it is possible to 
+            // specify the parameter `numBins', which controls the number of bins used in the approximation of the mutual information
+            // between features and label.
+
+            var pipeline = mlContext.Transforms.FeatureSelection.SelectFeaturesBasedOnMutualInformation(
+                outputColumnName: "FeaturesSelectedGroupB", inputColumnName: "GroupB", labelColumnName: "Label", slotsInOutput: 2);
+
+            // The pipeline can then be trained, using .Fit(), and the resulting transformer can be used to transform data. 
+            var transformedData = pipeline.Fit(data).Transform(data);
+
+            Console.WriteLine("Contents of column 'FeaturesSelectedGroupB'");
+            PrintDataColumn(transformedData, "FeaturesSelectedGroupB");
+            // Note, SelectFeaturesBasedOnMutualInformation retained 2 slots (out of 3).  
+            // 7 10
+            // 2 2
+            // 3 4
+            // 2 1
+
+            // Multi column example : This pipeline uses two columns for transformation
+            pipeline = mlContext.Transforms.FeatureSelection.SelectFeaturesBasedOnMutualInformation(
+                new InputOutputColumnPair[] { new InputOutputColumnPair("GroupB"), new InputOutputColumnPair("GroupC") },
+                labelColumnName: "Label",
+                slotsInOutput:4);
```

Comment:
>slotsInOutput:4 [](start = 16, length = 15)

one line comment about what this does.

Reply:
comment in line 28 should clarify this

---
In reply to: [271808888](https://github.com/dotnet/machinelearning/pull/3184#discussion_r271808888) [](ancestors = 271808888)